src/util/DeleteClassService.js

```diff
@@ -0,0 +1,52 @@
+import { deleteClass, getUserData } from './DataBaseRequests';
+
+import { useAuth } from '../AuthProvider';
+import { useState } from 'react';
+
+const useDeleteClass = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const { userData, setUserData } = useAuth();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const openModal = () => setIsModalOpen(true);
+  const closeModal = () => setIsModalOpen(false);
+
+  const handleDelete = async (token, classId, onRequestClose, onError, onSelectedClass, onSelectedId) => {
+    try {
+        setIsLoading(true);
+      let response = await deleteClass(token, classId);
+      if (response.status === 200) {
```

Comment:
You are handling an error correctly here but what if you get a non 200 response. That's not technically an error and also your loading state would be stuck here. You just need an else conditional to handle that non 200 response.

Reply:
got it.